db/view/view.hh

```diff
@@ -315,6 +316,10 @@ future<query::clustering_row_ranges> calculate_affected_clustering_ranges(
 
 bool needs_static_row(const mutation_partition& mp, const std::vector<view_and_base>& views);
 
+// Whether this node and shard should generate and send view updates for the given token.
+// Checks that the node is one of the replicas (not a pending replicas), and is ready for reads.
+bool should_generate_view_updates_on_this_shard(const schema_ptr& base, const locator::effective_replication_map_ptr& ermp, dht::token token);
```

Comment:
Nitpick: if you call this function in just one file it can be "static" in the file it uses and doesn't need to be in the header file at all.

Reply:
it's here because I thought the function belongs logically to the view interface, and it's placed close to `get_view_natural_endpoint`

Reply:
Ok. I don't feel strongly about it. I agree that more view-specific stuff should be in db/view and not in other places, so in that sense your decision is good. Too bad the caller is in one of the other places.